R/pipe_call_linter.R

```diff
@@ -33,12 +34,16 @@ pipe_call_linter <- function() {
     xml <- source_expression$xml_parsed_content
 
     bad_expr <- xml2::xml_find_all(xml, xpath)
+    pipe <- xml_text(xml_find_first(bad_expr, "preceding-sibling::SPECIAL[1]"))
 
     xml_nodes_to_lints(
       bad_expr,
       source_expression = source_expression,
-      lint_message = "Use explicit calls in magrittr pipes, i.e., `a %>% foo` should be `a %>% foo()`.",
+      lint_message =
+        sprintf("Use explicit calls in magrittr pipes, i.e., `a %1$s foo` should be `a %1$s foo()`.", pipe),
       type = "warning"
     )
   })
 }
+
+magrittr_pipes <- c("%>%", "%!>%", "%T>%", "%$%", "%<>%")
```

Comment:
Since it's re-used in other linters, this should probably go in a central file.

Reply:
I was looking around for such a file, seems we don't have one yet. Would apply to other internal constants as well, e.g.

https://github.com/r-lib/lintr/blob/bb7669afa4b90843e8358397e60d02e2fc512953/R/expect_type_linter.R#L76

https://github.com/r-lib/lintr/blob/bb7669afa4b90843e8358397e60d02e2fc512953/R/object_name_linter.R#L1

Follow-up, or I'll make that sweep in this PR?

Reply:
Follow up for the other constants to keep the diff smaller?

Reply:
#2048